src/lib/loadEventsNewsMetadataMap.ts

```diff
@@ -25,48 +30,35 @@ const queryNewsAndEvents = gql`
         }
         slug
         shortTitle
-        eventDescription
+        eventSummary
         eventDateAndTime
+        metaTitle
+        metaDescription
       }
     }
   }
 `;
 
-// generated query types are not returning
-type NewsAndEventsCollectionQuery = {
-  newsCollection: {
-    items: {
-      sys: { id: string };
-      __typename?: "News";
-      title?: string | null;
-      publicationDate?: string | null;
-      slug?: string | null;
-    }[];
-  };
-  eventEntryCollection: {
-    items: {
-      sys: {
-        id: string;
-      };
-      __typename?: "News";
-      title?: string | null;
-      slug?: string | null;
-      eventDateAndTime?: string | null;
-      eventDescription?: string | null;
-    }[];
-  };
-};
+type NewsSearchIndexingMetadataMapItem = NonNullable<
+  NewsAndEventsCollectionQuery["newsCollection"]
+>["items"][number];
+type EventSearchIndexingMetadataMapItem = NonNullable<
+  NewsAndEventsCollectionQuery["eventEntryCollection"]
+>["items"][number];
+
+type AdditionalMetadataFields = { url?: string | null; children?: string[] };
 
-// TODO: remove
-const constructEventSlug = (date: Date, slug: string): string =>
-  `${date.toISOString().split("T")[0]}-${slug}`;
+export type SearchIndexingMetadataMapItem =
+  | PageMetadataMapItem
+  | (NewsSearchIndexingMetadataMapItem & AdditionalMetadataFields)
+  | (EventSearchIndexingMetadataMapItem & AdditionalMetadataFields);
```

Comment:
This isn't perfect, but it does give us what we need to ensure we're not querying for fields that don't exist. The downside here is that we need to check the `__typename` for the fields that don't overlap, but I think that's a fair trade-off and makes it easier to reason about.